### PR TITLE
Remove objectExists() from readJson

### PR DESCRIPTION
**Pull request by tischi**
Does that look good?
Could you merge this today?

### CODE REVIEW

File: src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java [hunk 1]
```diff
@@ -261,42 +261,23 @@ public DataBlock<?> readBlock(
...
-                }
-
+        try {
+            try (final InputStream in = this.readS3Object(objectPath)) {
+                return GsonAttributesParser.readAttributes(new InputStreamReader(in), gson);
             }
-
-            return var5;
+        } catch (AmazonS3Exception ase) {
+            if ("NoSuchKey".equals(ase.getErrorCode())) {
```
Comment (constantinpape) [inline src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java:277]: Minor comment: but could you write this the other way round? `errorCode.equals("NoSuchKey")` makes more sense logically and would help to understand this.

File: src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java [hunk 2]
```diff
@@ -261,42 +261,23 @@ public DataBlock<?> readBlock(
...
-                        } catch (Throwable var13) {
-                            var4.addSuppressed(var13);
-                        }
-                    } else {
-                        in.close();
-                    }
-                }
-
+        try {
+            try (final InputStream in = this.readS3Object(objectPath)) {
```
Comment (constantinpape) [inline src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java:273]: What's the reason for the nested `try` here? Can't we just have a single one?

**Review by constantinpape — COMMENTED** [review]
Looks good overall, only two comments on code readability.

File: src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java (hunk 2 above)
Comment (tischi) [reply src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java]: No clue. I removed the nesting.

File: src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java (hunk 1 above)
Comment (tischi) [reply src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java]: Good point.

**Comment by tischi** [comment]
@constantinpape 
Thanks for the suggestions!!
Better now?

**Comment by tischi** [comment]
@constantinpape @KateMoreva 
Maybe we first merge this: https://github.com/mobie/mobie-io/pull/96
And then this one on top?

**Review by KateMoreva — APPROVED** [review]

**Review by constantinpape — APPROVED** [review]
Loks good to me now

**Pull request merged**